src/rgw/rgw_rest_swift.cc

```diff
@@ -802,12 +802,12 @@ static int get_swift_account_settings(req_state * const s,
   const char * const acl_attr = s->info.env->get("HTTP_X_ACCOUNT_ACCESS_CONTROL");
   if (acl_attr) {
     RGWAccessControlPolicy_SWIFTAcct swift_acct_policy(s->cct);
```

Comment:
`-EINVAL` should be mapped to `400 Bad Request` which is completely OK according to what Swift is doing:

```
$ curl -i -H "X-Auth-Token: $token" "$publicURL" -X POST -H "X-Account-Access-Control: bla"
HTTP/1.1 400 Bad Request
Content-Length: 76
Content-Type: text/plain; charset=UTF-8
X-Trans-Id: txe731606404814e4fbefab-0057b71d87
Date: Fri, 19 Aug 2016 14:53:59 GMT

X-Account-Access-Control invalid: Syntax error in input ('bla')

Input: bla
```

Without your patch RadosGW returns `204 No Content`. This behavior looks like a bug. I'm marking this PR as *bug fix*.
```
HTTP/1.1 204 No Content
X-Trans-Id: tx000000000000000000006-0057ba1fe9-1012-default
Content-Type: text/plain; charset=utf-8
Date: Sun, 21 Aug 2016 21:40:57 GMT
```